### PR TITLE
@orta: Fixes for Kiosk CC testing on Staging, Unresolved bids, Bidding errors

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes:

- Testing Kiosks on staging
- Unresolved bids messaging
- [API errors resulting in being outbid](https://trello.com/c/tUeKka0c/70-kiosk-should-not-show-bidding-error-for-successful-bids-or-outbids)

### CODE REVIEW

**Comment by ashfurrow** [comment]
- [ ] Add Stripe's test publishable key.
- [ ] Add CardFlight's API key and merchant account.
- [ ] Write code to use one or the other against the API depending on app settings.
- [ ] Test against staging with Stripe's test CC's.

File: Kiosk/Bid Fulfillment/BidderNetworkModel.swift [hunk 1]
```diff
@@ -36,8 +36,12 @@ class BidderNetworkModel: NSObject {
     }
 
     private func createOrUpdateUser() -> RACSignal {
-        let boolSignal = self.checkUserEmailExists(fulfillmentController.bidDetails.newUser.email!)
+        // Signal to test for user existence (does a user exist with this email?)
+        let boolSignal = self.checkUserEmailExists(fulfillmentController.bidDetails.newUser.email ?? "")
+        // If the user exists, update their info to the API, otherwise create a new user.
         let signal = RACSignal.`if`(boolSignal, then: self.updateUser(), `else`: self.createNewUser())
+
+        // After update/create signal finishes, add a CC to their account (if we've collceted one)
```
Comment (orta) [inline Kiosk/Bid Fulfillment/BidderNetworkModel.swift]: typo: collected

File: Kiosk/Bid Fulfillment/BidderNetworkModel.swift [hunk 2]
```diff
@@ -66,19 +70,22 @@ class BidderNetworkModel: NSObject {
 
         return updateProviderIfNecessary().then {
             self.fulfillmentController.loggedInProvider!.request(endpoint).filterSuccessfulStatusCodes().mapJSON()
-        }.doError { (error) in
+        }.logNext().doError { (error) in
```
Comment (orta) [inline Kiosk/Bid Fulfillment/BidderNetworkModel.swift]: nice

**Comment by orta** [comment]
structurally it looks good, but there's a lot of code in here with changes and no tests

**Comment by ashfurrow** [comment]
Cool, I'll add some tests.

File: Kiosk/App/Views/SimulatorOnlyView.swift [hunk 3]
```diff
@@ -3,8 +3,13 @@ import UIKit
 class DeveloperOnlyView: UIView {
 
     override func awakeFromNib() {
-        let shouldShow = AppSetup.sharedState.showDebugButtons && AppSetup.sharedState.useStaging
-        let notTests = NSClassFromString("XCTest") == nil
-        self.hidden = !shouldShow && notTests
```
Comment (ashfurrow) [inline Kiosk/App/Views/SimulatorOnlyView.swift]: The logic here was messy, so I made it prettier.

**Comment by ashfurrow** [comment]
OK, got the tests passing again, and added some minimal tests. `ConfirmYourBidPINViewController` needs to be split into a view model, I've opened https://github.com/artsy/eidolon/issues/511 to do so when I get a chance. 

The dev-only views were showing in some tests, which has been correct. Also, I changed the placeholder for the manual CC entry's year to be in the future.

File: KioskTests/Bid Fulfillment/PlaceBidNetworkModelTests.swift [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                subject.bidSignal().subscribeCompleted {
+                    done()
+                }
+            }
+
+            // ID retrieved from CreateABid.json
+            expect(subject.bidderPosition?.id) == "5437dd107261692daa170000"
+        }
+
+        it("maps bid deatils into a proper request") {
```
Comment (orta) [inline KioskTests/Bid Fulfillment/PlaceBidNetworkModelTests.swift]: chore: typo

File: Podfile [hunk 5]
```diff
@@ -10,9 +10,12 @@ plugin 'cocoapods-keys', {
...
     'MixpanelStagingAPIClientKey',
-    'CardflightAPIClientKey',
-    'CardflightMerchantAccountToken',
-    'StripePublishableKey'
+    'CardflightProductionAPIClientKey',
+    'CardflightProductionMerchantAccountToken',
+    'StripeProductionPublishableKey',
+    'CardflightStagingAPIClientKey',
+    'CardflightStagingMerchantAccountToken',
+    'StripeStagingPublishableKey'
```
Comment (orta) [inline Podfile]: `make oss` will probably need an update too

**Comment by orta** [comment]
Blocker, this test seems wrong, either in the logic or the name
screen shot 2015-09-24 at 07 13 09

**Comment by ashfurrow** [comment]
No, that's correct. The bid has been submitted, we just don't know whether or not the user is the highest bidder. "Bid Submitted" is the most neutral language we could find. The previous screenshot was incorrect, as well, since it indicates you need to increase your bid (the test setup was wrong).

**Comment by ashfurrow** [comment]
I'll rename the test, remove "error" – good idea.

File: Podfile (hunk 5 above)
Comment (ashfurrow) [inline Podfile]: We don't have a Makefile; both OSS users and CI use this lane, so we should be good.

**Comment by orta** [comment]
Ace :+1:

**Pull request merged**